### PR TITLE
[SHLWAPI_APITEST] Enhance AssocQueryString testcase

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Add the tests of "Tricky Pointer" (the pointer that is non-null and `IS_INTRESOURCE(...)` is true).
JIRA issue: [CORE-20467](https://jira.reactos.org/browse/CORE-20467)

## Proposed changes

- Directly call `AssocQueryStringA/W` functions without using `GetProcAddress`.
- Use `ASSOCF_NONE` instead of zero.
- Add the tests of "Tricky Pointers" handling.

## Screenshots

WinXP:
WinXP
Successful.

Win10 x64:
Win10
Successful.

## Testbot runs (Filled in by Devs)

- [x] KVM x86: https://reactos.org/testman/compare.php?ids=107922,107954
- [x] KVM x64: https://reactos.org/testman/compare.php?ids=107923,107955
- [x] Win2003 x64: https://reactos.org/testman/compare.php?ids=107921,107953

### CODE REVIEW

**Comment by katahiromz** [comment]
BEFORE: 628fc9d
AFTER: 2493023

KVM x86: https://reactos.org/testman/compare.php?ids=107922,107954
KVM x64: https://reactos.org/testman/compare.php?ids=107923,107955
Win2003 x64: https://reactos.org/testman/compare.php?ids=107921,107953

**Comment by whindsaks** [comment]
This is a nice lazy way if you don't care about the output length, shame they removed it in Vista.

File: modules/rostests/apitests/shlwapi/AssocQueryString.c [hunk 1]
```diff
@@ -161,27 +143,59 @@ static void TEST_AssocQueryStringW(void)
...
+    if (bVistaPlus)
+        ok_long(hr, 0xDEADFACE);
+    else
+        ok_long(hr, HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND));
+
+    lstrcpynW(szPathW, s_szTextFileW, _countof(szPathW));
+    cch = _countof(szOutputW);
+    _SEH2_TRY
+    {
+        hr = AssocQueryStringW(ASSOCF_NONE, ASSOCSTR_EXECUTABLE, szPathW, NULL, szPathW, (PDWORD)UlongToPtr(cch));
```
Comment (whindsaks) [inline modules/rostests/apitests/shlwapi/AssocQueryString.c]: `szOutputW`?

Comment (katahiromz) [reply modules/rostests/apitests/shlwapi/AssocQueryString.c]: Fixed.

**Review by whindsaks — APPROVED** [review]
